javascript/ql/src/semmle/javascript/AST.qll

```diff
@@ -181,51 +198,118 @@ class TopLevel extends @toplevel, StmtContainer {
 
 /**
  * A stand-alone file or script originating from an HTML `<script>` element.
+ *
+ * Example:
+ *
+ * ```
+ * <script>
+ * window.done = true;
+ * </script>
+ * ```
  */
 class Script extends TopLevel {
   Script() { this instanceof @script or this instanceof @inline_script }
 }
 
 /**
  * A stand-alone file or an external script originating from an HTML `<script>` element.
+ *
+ * Example:
+ *
+ * ```
+ * #! /usr/bin/node
```

Comment:
Do you use the `#!` to indicate that this is an entire file and not just a statement?
If not, do we really need `#!` in the example?
It is strictly speaking not valid JavaScript, and it wont make sense to include through the `<script>` element mentioned above.

Reply:
Yes, the idea is to show different kinds of scripts. I don't think there is an implication that one would include it using a `<script>` element?